libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Analytics/SFSDKAILTNPublisher.m

```diff
@@ -53,41 +51,32 @@ + (BOOL) publish:(NSArray *) events {
-        NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:loggingEndpointUrl
-                                                               cachePolicy:NSURLRequestReloadIgnoringCacheData
-                                                           timeoutInterval:60.0];
-        [request setHTTPMethod:@"POST"];
-        [request setValue:userAgent forHTTPHeaderField:@"User-Agent"];
-        [request setValue:@"false" forHTTPHeaderField:@"X-Chatter-Entity-Encoding"];
-        [request setValue:token forHTTPHeaderField:@"Authorization"];
-        [request setValue:@"application/json" forHTTPHeaderField:@"Content-Type"];
+    NSString *path = [NSString stringWithFormat:@"/%@/%@", kSFRestDefaultAPIVersion, kRestApiSuffix];
+    SFRestRequest *request = [SFRestRequest requestWithMethod:SFRestMethodPOST path:path queryParams:nil];
 
-        // Adds GZIP compression.
-        NSString *bodyString = [[self class] dictionaryAsJSONString:bodyDictionary];
-        NSData *bodyData = [bodyString dataUsingEncoding:NSUTF8StringEncoding];
-        NSData *postData = [bodyData gzipDeflate];
-        [request setValue:@"gzip" forHTTPHeaderField:@"Content-Encoding"];
-        [request setValue:[NSString stringWithFormat:@"%lu", (unsigned long)[postData length]] forHTTPHeaderField:@"Content-Length"];
-        [request setHTTPBody:postData];
-        NSHTTPURLResponse* response = nil;
-        [NSURLConnection sendSynchronousRequest:request returningResponse:&response error:error];
-        return response;
-    };
-    NSHTTPURLResponse *response = makeSynchronousRequest(&error);
-    NSInteger code = [response statusCode];
-    if (error) {
-        [SFLogger log:[self class] level:SFLogLevelError format:@"Upload failed %ld %@", (long)[error code], [error localizedDescription]];
-        return NO;
-    } else if (code >= 200 && code < 300) {
-        return YES;
+    // Adds GZIP compression.
+    NSString *bodyString = [[self class] dictionaryAsJSONString:bodyDictionary];
+    NSData *bodyData = [bodyString dataUsingEncoding:NSUTF8StringEncoding];
+    NSData *postData = [bodyData gzipDeflate];
+    [request setCustomRequestBodyData:postData contentType:@"application/json"];
+    [request setHeaderValue:@"gzip" forHeaderName:@"Content-Encoding"];
+    [request setHeaderValue:[NSString stringWithFormat:@"%lu", (unsigned long)[postData length]] forHeaderName:@"Content-Length"];
+    __block BOOL finished = NO;
+    __block BOOL success = NO;
+    [[SFRestAPI sharedInstance] sendRESTRequest:request failBlock:^(NSError *e) {
+        finished = YES;
+            if (e) {
+                [SFLogger log:[self class] level:SFLogLevelError format:@"Upload failed %ld %@", (long)[e code], [e localizedDescription]];
+            }
+        } completeBlock:^(id response) {
+            finished = YES;
+            success = YES;
+        }
+    ];
+    while(!finished) {
```

Comment:
Turning a network request into a synchronous block is an anti-pattern, and should have a compelling argument for why it shouldn't be orchestrated in an asynchronous design pattern.  Looking through the code, I'm not seeing that argument.

Why are we running synchronous network requests here?